[SPARK-25683][Core] Updated the log for the firstTime event Drop occurs

shivusondur opened the pull request:
## What changes were proposed in this pull request?

When the first dropEvent occurs, LastReportTimestamp was printing in the log as
Wed Dec 31 16:00:00 PST 1969
(Dropped 1 events from eventLog since Wed Dec 31 16:00:00 PST 1969.)

The reason is that lastReportTimestamp initialized with 0.

Now log is updated to print "... since the application starts" if 'lastReportTimestamp' == 0.
this will happens first dropEvent occurs.

## How was this patch tested?
Manually verified.

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

shivusondur commented:
cc @HyukjinKwon ,  @vanzin

shivusondur commented on core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala:
```diff
@@ -159,6 +159,11 @@ private class AsyncEventQueue(
+      // If first time the dropEvent occurs, then record current time as the
+      // lastReportTimestamp.
+      if (lastReportTimestamp == 0) {
+        lastReportTimestamp = System.currentTimeMillis()
```
Yes i will do

jiangxb1987 commented:
Though it looks a little strange, the log content is actually right, I don't think we want to make the last report timestamp to current time (that can confuse users what happened before that timestamp).

shivusondur commented:
@jiangxb1987 
Thanks for your comment,
i think printing "since Wed Dec 31 16:00:00 PST 1969" still looks strange,
Instead we can print "**since start of the application** for first time event Dropping, this looks more appropriate.

so first time log should look like this 
**18/10/08 17:51:40 WARN AsyncEventQueue: Dropped 1 events since start of the application**

Instead of 
**18/10/08 17:51:40 WARN AsyncEventQueue: Dropped 1 events from eventLog since Wed Dec 31 16:00:00 PST 1969.**

please correct me if i am wrong.

shivusondur committed:
[SPARK-25683] Make AsyncEventQueue.lastReportTimestamp inital value as the currentTime instead of 0

shivusondur commented:
@jiangxb1987 @dongjoon-hyun 
I have modified accordingly.

Tested locally with existing UT (to make drop the event, locally "LISTENER_BUS_EVENT_QUEUE_CAPACITY" is set to 1), using println.
Tested as in below snap. 
![screenshot from 2018-10-15 14-51-48](https://user-images.githubusercontent.com/7912929/46943354-1967cf80-d08d-11e8-8ffc-4c6d536a077b.png)

vanzin commented:
ok to test

vanzin commented:
The PR title is not accurate anymore.

SparkQA commented:
**[Test build #97412 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97412/testReport)** for PR 22677 at commit [`d71c74f`](https://github.com/apache/spark/commit/d71c74f3d8a8b22ac79474ad864606d76bbd69ba).

SparkQA commented:
**[Test build #97412 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97412/testReport)** for PR 22677 at commit [`d71c74f`](https://github.com/apache/spark/commit/d71c74f3d8a8b22ac79474ad864606d76bbd69ba).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97412/
Test FAILed.

shivusondur committed:
[SPARK-25683] Make AsyncEventQueue.lastReportTimestamp inital value as the currentTime instead of 0

shivusondur committed:
[SPARK-25683] Make AsyncEventQueue.lastReportTimestamp inital value as the currentTime instead of 0

SparkQA commented:
**[Test build #97435 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97435/testReport)** for PR 22677 at commit [`4b36af3`](https://github.com/apache/spark/commit/4b36af375e28154e0f978416bd4ffe9abeda2bda).

SparkQA commented:
**[Test build #97437 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97437/testReport)** for PR 22677 at commit [`7105cce`](https://github.com/apache/spark/commit/7105cce2657e81ed118c4713b517c89abaa7a9f2).

SparkQA commented:
**[Test build #97437 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97437/testReport)** for PR 22677 at commit [`7105cce`](https://github.com/apache/spark/commit/7105cce2657e81ed118c4713b517c89abaa7a9f2).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

SparkQA commented:
**[Test build #97435 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97435/testReport)** for PR 22677 at commit [`4b36af3`](https://github.com/apache/spark/commit/4b36af375e28154e0f978416bd4ffe9abeda2bda).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97437/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97435/
Test FAILed.

shivusondur committed:
nit: modication

SparkQA commented:
**[Test build #97446 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97446/testReport)** for PR 22677 at commit [`3e3ad24`](https://github.com/apache/spark/commit/3e3ad2400de8ff5408a949736319409206a3e9b3).

SparkQA commented:
**[Test build #97446 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97446/testReport)** for PR 22677 at commit [`3e3ad24`](https://github.com/apache/spark/commit/3e3ad2400de8ff5408a949736319409206a3e9b3).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97446/
Test PASSed.

shivusondur commented:
Hi, 
I have modified the title and description according to code change. Build also passed. 
Kindly check and merge.

shivusondur commented:
cc @vanzin

dongjoon-hyun commented:
Retest this please.

SparkQA commented:
**[Test build #97516 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97516/testReport)** for PR 22677 at commit [`3e3ad24`](https://github.com/apache/spark/commit/3e3ad2400de8ff5408a949736319409206a3e9b3).

dongjoon-hyun commented on core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala:
```diff
@@ -169,7 +169,8 @@ private class AsyncEventQueue(
           val previous = new java.util.Date(prevLastReportTimestamp)
-          logWarning(s"Dropped $droppedCount events from $name since $previous.")
+          logWarning(s"Dropped $droppedCount events from $name since " +
+            s"${if (prevLastReportTimestamp == 0) "start of the application" else s"$previous"}.")
```
nit. "start of the application" -> "the application starts", @shivusondur ?

shivusondur replied:
@dongjoon-hyun Done.

dongjoon-hyun commented:
@shivusondur . It seems that your PR almost finished. Please update JIRA title and description together. https://issues.apache.org/jira/browse/SPARK-25683

SparkQA commented:
**[Test build #97516 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97516/testReport)** for PR 22677 at commit [`3e3ad24`](https://github.com/apache/spark/commit/3e3ad2400de8ff5408a949736319409206a3e9b3).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97516/
Test FAILed.

SparkQA commented:
**[Test build #97518 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97518/testReport)** for PR 22677 at commit [`b1f2e48`](https://github.com/apache/spark/commit/b1f2e48ae8ba0ff597c3b3adf0fb0e279b94244f).

SparkQA commented:
**[Test build #97518 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97518/testReport)** for PR 22677 at commit [`b1f2e48`](https://github.com/apache/spark/commit/b1f2e48ae8ba0ff597c3b3adf0fb0e279b94244f).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97518/
Test FAILed.

shivusondur committed:
Update AsyncEventQueue.scala

shivusondur committed:
Update AsyncEventQueue.scala

SparkQA commented:
**[Test build #97532 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97532/testReport)** for PR 22677 at commit [`df21621`](https://github.com/apache/spark/commit/df216217512f5745667c75993c9ede0c492b8e34).

SparkQA commented:
**[Test build #97532 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97532/testReport)** for PR 22677 at commit [`df21621`](https://github.com/apache/spark/commit/df216217512f5745667c75993c9ede0c492b8e34).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97532/
Test PASSed.

dongjoon-hyun reviewed (approved):
+1, LGTM.

vanzin commented on core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala:
```diff
@@ -169,7 +169,8 @@ private class AsyncEventQueue(
           val previous = new java.util.Date(prevLastReportTimestamp)
-          logWarning(s"Dropped $droppedCount events from $name since $previous.")
+          logWarning(s"Dropped $droppedCount events from $name since " +
+            s"${if (prevLastReportTimestamp == 0) "the application starts" else s"$previous"}.")
```
Actually the previous version was grammatically correct. Either:

- since the application started
- since the start of the application

But not the current version...

shivusondur replied:
@vanzin 
Thanks for your comment.
Updated code according to your suggestion.

dongjoon-hyun replied:
Oops. I missed the past tense.

shivusondur committed:
Update AsyncEventQueue.scala

SparkQA commented:
**[Test build #97553 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97553/testReport)** for PR 22677 at commit [`0f2b2d1`](https://github.com/apache/spark/commit/0f2b2d144ed2f44d30f01515516cc25d3b214cb2).

vanzin commented:
Merging to master.

SparkQA commented:
**[Test build #97553 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97553/testReport)** for PR 22677 at commit [`0f2b2d1`](https://github.com/apache/spark/commit/0f2b2d144ed2f44d30f01515516cc25d3b214cb2).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97553/
Test PASSed.
